### PR TITLE
Tsc 336 create diagram for login flow

### PR DESCRIPTION
**Pull request by asivath**
Updated documentation for the login routes

### CODE REVIEW

File: assets/Database-Structure.drawio.png
Comment (paologum) [inline assets/Database-Structure.drawio.png:1]: can you add a quick description of what verify, password, and invalidate means in the verification table

File: assets/Login-Flows.drawio.png
Comment (paologum) [inline assets/Login-Flows.drawio.png:1]: what does password verification on the left mean i forgot

Comment (paologum) [inline assets/Login-Flows.drawio.png:1]: one thing that confused me was that we say email sent for avoiding attacks on user existence, but on signup don't we tell them that a user already exists with that email? isn't that the same thing and might provoke attacks?

Comment (asivath) [reply assets/Login-Flows.drawio.png:1]: the first verification is checking that the user inputs are valid, so not empty. the second one is checking if the password is correct. I'm going to replace the first one with "Validate User Inputs".

Comment (asivath) [reply assets/Login-Flows.drawio.png:1]: it is based on username or email matching, but yes its unsecure. thats why we need captcha/rate limiting. we could also just have a vague error message for everything.

**Review by paologum — APPROVED** [review]

**Pull request merged**